index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
+                                          args.concat(file),
+                                          {stdio: stdio});
+            clangFormatProcess.on('close', function(exit) {
+              if (exit !== 0) callback(exit);
+              else callback(null, exit);
+            });
+          };
+        }),
+        200,
```

Comment:
Setup is 8gb ram, i7-4500U CPU @ 1.8GHz.

Ran `clang-format -i --glob=node_modules/**/*.js` on a `angular-cli` proj. Didn't notice much ram consumption, but ate through the CPU like mad. In fact, no more than 5-6 processes seem to be running at any time. And then it had a segfault after a while.

Toned it down to 2 processes at once, still segfaulted.

Tried a folder with a single 6meg js file, took (roughly) 45s and used up 30% of my CPU and 350-450mb ram. No segfault. Left behind a `.TMP` file for whatever reason.

Same thing but copied the file 30 times, still only 2 processes. Took some 17 minutes, no segfault, similar resource consumption.

So tbh if CPU consumption can get this high, I don't think we should have more than 2-4 processes at the same time.